app/gui/src/model/suggestion_database.rs

```diff
@@ -596,4 +668,203 @@ mod test {
         assert_eq!(db.lookup(3).unwrap().arguments[2].name, "NewArg");
         assert_eq!(db.version.get(), 8);
     }
+
+    fn expect_lookup(db: &SuggestionDatabase, fully_qualified_name: &str) {
+        let lookup = db.lookup_by_fully_qualified_name(fully_qualified_name);
+        assert!(lookup.is_some());
+        let name = fully_qualified_name.rsplit('.').next().unwrap();
+        assert_eq!(lookup.unwrap().name, name);
+    }
+
+    fn expect_no_lookup(db: &SuggestionDatabase, fully_qualified_name: &str) {
+        let lookup = db.lookup_by_fully_qualified_name(fully_qualified_name);
+        assert_eq!(lookup, None);
+    }
+
+    fn db_entry(id: SuggestionId, suggestion: SuggestionEntry) -> SuggestionsDatabaseEntry {
+        SuggestionsDatabaseEntry { id, suggestion }
+    }
+
+    #[test]
+    fn lookup_by_fully_qualified_name_in_db_created_from_ls_response() {
```

Comment:
As I've written in previous PR review - this thing is big and does not have docs explaining what it is testing. Reading 50-100 lines to understadn what test does is a bad idea.

Reply:
I like the new comments! @farmaazon hmm, when reading this name when doing code review it was not obvious. Without it explanation it would probably be not obvious for me now neither. But the docs do the job now. Plus, the code is way easier to read now!